pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -588,14 +590,27 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 
 	// determine if we can gather info for all resources
 	infos := []hiveresource.Info{}
+	unstructureds := []*unstructured.Unstructured{}
```

Comment:
```suggestion
	unstructureds := make([]*unstructured.Unstructured, len(resources))
```

And later, instead of the append,
```
unstructureds[i] = u
```